### DISCUSSION TITLE
One Code-Workspace file for both Windows and Linux

### DISCUSSION THREAD

**Discussion by trolley78** [VSCodium/vscodium#1837] [Q&A]
Hi,

I am trying to get one code-workspace file that I can open for Windows and Linux.  I do most of my development thru Windows.  When not in my office, I use a Linux laptop.  I want to set Folders up to have the Windows folders and the Linux folders.  They point to the same place.  I use putty to connect to a SCO OpenServer 5 system to do my compiles using RM/COBOL.  I know this is old school, but I still have a client running on SCO OpenServer 5.

As I said looking to put all my settings for the workspace into one file.  In this way, I do not have to keep comparing the workspace files to each other.  Any help would be greatly appreciated.

Regards,

Rich Di Iulio

**Comment by Jeong-ilKajokaya** [comment]
Hi Rich,

You can definitely create a single `.code-workspace` file that works on both Windows and Linux, but there are some considerations for cross-platform paths.

---

### 1. Using OS-specific Paths

VSCode/VSCodium allows you to define folders in a workspace file using absolute paths. Since Windows and Linux use different path formats (`C:\path\to\folder` vs `/home/user/path`), you can structure the workspace file to include **both sets of paths** and let your OS ignore the ones that don’t exist.

Example `myworkspace.code-workspace`:

```json
{
  "folders": [
    {
      "path": "C:\\Users\\YourName\\Project",
      "name": "Windows Project"
    },
    {
      "path": "/home/username/Project",
      "name": "Linux Project"
    }
  ],
  "settings": {
    // workspace-specific settings here
  }
}
```

* Only the paths that exist on the current OS will appear in Explorer.
* The workspace will ignore folders that don’t exist.

---

### 2. Remote Development via SSH (optional)

Since you are connecting to SCO OpenServer, you might also consider using **Remote SSH extension**:

* This allows you to define a host (`SCO OpenServer`) and open it as a remote workspace.
* You can keep your local Windows/Linux workspace the same, and VSCode handles the remote connection transparently.

---

### 3. Tips

* Keep workspace-specific settings (like keybindings, extensions, or terminal paths) minimal to avoid OS-specific conflicts.
* Use relative paths if possible within the project directory, as they are OS-agnostic.
* Save the workspace file in a location that is accessible from both Windows and Linux (e.g., cloud storage or version control).

**Summary:**

* Include both Windows and Linux paths in the `folders` array; VSCode will ignore paths that don’t exist.
* Use Remote SSH for SCO OpenServer if you want to integrate remote builds.
* Keep workspace settings OS-neutral as much as possible.
